sklearn/utils/estimator_checks.py

```diff
@@ -131,6 +131,30 @@ def _yield_classifier_checks(name, Classifier):
     if 'class_weight' in Classifier().get_params().keys():
         yield check_class_weight_classifiers
 
+def check_supervised_y_no_nan(name, Estimator):
+    # Checks that the Estimator targets are not NaN.
+
+    warnings.simplefilter("ignore")
```

Comment:
Why are there warnings? If it's due to the division by zero, you can replace it with y1 = np.inf * np.ones(10)